clarifai/client/search.py

```diff
@@ -33,23 +36,34 @@ def __init__(self,
         app_id (str): App ID.
         top_k (int, optional): Top K results to retrieve. Defaults to 10.
         metric (str, optional): Similarity metric (either 'cosine' or 'euclidean'). Defaults to 'cosine'.
+        alogrithm (str, optional): Search algorithm (either 'nearest_neighbor' or 'brute_force'). Defaults to 'nearest_neighbor'.
+        pagination (bool, optional): Enable pagination. Defaults to False.
         base_url (str, optional): Base API url. Defaults to "https://api.clarifai.com".
         pat (str, optional): A personal access token for authentication. Can be set as env var CLARIFAI_PAT
         token (str): A session token for authentication. Accepts either a session token or a pat. Can be set as env var CLARIFAI_SESSION_TOKEN
         root_certificates_path (str): Path to the SSL root certificates file, used to establish secure gRPC connections.
 
     Raises:
         UserError: If the metric is not 'cosine' or 'euclidean'.
+        UserError: If the algorithm is not 'nearest_neighbor' or 'brute_force'.
     """
     if metric not in ["cosine", "euclidean"]:
       raise UserError("Metric should be either cosine or euclidean")
+    if algorithm not in ["nearest_neighbor", "brute_force"]:
+      raise UserError("Algorithm should be either nearest_neighbor or brute_force")
```

Comment:
Should we have a condition here to raise if they've given incorrect search params ?
eg: `cosine` with `nearest_neighbor`

```
// Metric used for search. Can be EUCLIDEAN_DISTANCE (default) or COSINE_DISTANCE.
  // Currently only brute force search supports non-eudlicean metrics.
  enum Metric {
    METRIC_NOT_SET = 0;
    EUCLIDEAN_DISTANCE = 1;
    COSINE_DISTANCE = 2;
  }
  Metric metric = 13;
```